.github/workflows/e2e.yml

```diff
@@ -1,5 +1,5 @@
 name: e2e
-on: [push, pull_request]
```

Comment:
Why remove 'push'?

Reply:
I notice that two jobs were created for the virtual tests even if I only pushed one commit to the branch. I assumed that the problem was that the workflow was executed twice: one for push event and one for pull_request (synchronized) event.

Reply:
I see. This is an issue. The curious thing is when they continue adding to the PR, I assume it will not trigger! I think maybe just leaving push and removing pull_request, we catch all scenarios! WDYT?

Reply:
I dig more about those event types, and in my opinion **push** event does no contain information about labels. It provides less information than **pull_request** event type.

In the logs from **pull_request** there is an array like below
```
 "labels": [
        {
          "color": "A8EE42",
          "default": false,
          "description": "",
          "id": 2899356542,
          "name": "hardware",
          "node_id": "MDU6TGFiZWwyODk5MzU2NTQy",
          "url": "https://api.github.com/repos/MichalGuzieniuk/network-resources-injector/labels/hardware"
        }
      ],
```

to see all data that is provided within even we can use dump github object and display it while executing workflow. Code at https://github.com/MichalGuzieniuk/network-resources-injector/pull/2/files

```
- name: Dump GitHub context
        env:
          GITHUB_CONTEXT: ${{ toJson(github) }}
        run: echo "$GITHUB_CONTEXT"
```

Now, if you will push another commit to already open PR that is labeled with some specific label for instance _hardware_, it will run once again tests against newest changes.